### PR TITLE
[inductor] Add test for debug.trace mode

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #113240

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: torch/_inductor/scheduler.py [hunk 1]
```diff
@@ -107,7 +107,7 @@ def debug_str(self) -> str:
         """Longer form printout for trace logs"""
         name = self.get_name()
         lines = [
-            f"{name}: {type(self).__name__}({type(self.node).__name__})",
+            f"{name}: {type(self).__name__}({type(getattr(self, 'node', None)).__name__})",
```
Comment (oulgen) [inline torch/_inductor/scheduler.py:110]: Why did you need this? Shouldn't we go other way around and always define node?

**Review by oulgen — APPROVED** [review]

File: torch/_inductor/scheduler.py (hunk 1 above)
Comment (jansel) [reply torch/_inductor/scheduler.py:110]: See thread on slack:
https://pytorch.slack.com/archives/C033H6DJSJU/p1699417275970679?thread_ts=1699412125.335719&cid=C033H6DJSJU

I wasn't able to repro the attribute error from slack here, but this should be a safe change.
